configure.ac

```diff
@@ -917,9 +941,9 @@ if test "${enable_backend_prometeus_remote_write}" != "no" -a "${have_libprotobu
                                                            -a "${have_protoc}" = "yes" -a "${have_CXX_compiler}" = "yes"; then
     enable_backend_prometheus_remote_write="yes"
     AC_DEFINE([ENABLE_PROMETHEUS_REMOTE_WRITE], [1], [Prometheus remote write API usability])
-    OPTIONAL_PROMETHEUS_REMOTE_WRITE_CFLAGS="${PROTOBUF_CFLAGS} ${SNAPPY_CFLAGS}"
+    OPTIONAL_PROMETHEUS_REMOTE_WRITE_CFLAGS="${PROTOBUF_CFLAGS}"
     CXX11FLAG="-std=c++11"
-    OPTIONAL_PROMETHEUS_REMOTE_WRITE_LIBS="${PROTOBUF_LIBS} ${SNAPPY_LIBS} "
+    OPTIONAL_PROMETHEUS_REMOTE_WRITE_LIBS="${PROTOBUF_LIBS} -lsnappy"
```

Comment:
Why not simply populate SNAPPY_CFLAGS and SNAPPY_LIBS respectively?

SNAPPY_LIBS="${SNAPPY_LIBS} -lsnappy"
and
SNAPPY_CFLAGS="${SNAPPY_CFLAGS}"
respectively

Reply:
They were populated by the `AC_CHECK_LIB` automatically. Do we really need them now?

Reply:
Just to maintain compatibility with what it suppose to happen.
we only want to override pkg-config failing to provide us with the package information